src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/letter/InternalNonStandardDirectionLetterTemplate.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.reform.iacasedocumentsapi.domain.templates.letter;
+
+import static uk.gov.hmcts.reform.iacasedocumentsapi.domain.utils.AsylumCaseUtils.getAppellantPersonalisation;
+import static uk.gov.hmcts.reform.iacasedocumentsapi.domain.utils.DateUtils.formatDateForNotificationAttachmentDocument;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.Direction;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.DirectionTag;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.CaseDetails;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.templates.DocumentTemplate;
+import uk.gov.hmcts.reform.iacasedocumentsapi.infrastructure.CustomerServicesProvider;
+import uk.gov.hmcts.reform.iacasedocumentsapi.infrastructure.DirectionFinder;
+
+@Component
+public class InternalNonStandardDirectionLetterTemplate implements DocumentTemplate<AsylumCase> {
+
+    private final String templateName;
+    private final CustomerServicesProvider customerServicesProvider;
+    private final DirectionFinder directionFinder;
+
+    public InternalNonStandardDirectionLetterTemplate(
+        @Value("${internalNonStandardDirectionLetter.templateName}") String templateName,
+        CustomerServicesProvider customerServicesProvider,
+        DirectionFinder directionFinder) {
+        this.templateName = templateName;
+        this.customerServicesProvider = customerServicesProvider;
+        this.directionFinder = directionFinder;
+    }
+
+    public String getName() {
+
+        return templateName;
+    }
+
+    public Map<String, Object> mapFieldValues(
+        CaseDetails<AsylumCase> caseDetails
+    ) {
+        final AsylumCase asylumCase = caseDetails.getCaseData();
+
+        final Map<String, Object> fieldValues = new HashMap<>();
+
+        Optional<Direction> direction = directionFinder.findFirst(asylumCase, DirectionTag.NONE);
```

Comment:
This will fail sonarQ. We must do an isPresent (or isEmpty) check on all optional values. If its not present, throw an exception. If its present, then do a .get().

Reply:
The other comments are just for your knowledge and you dont HAVE TO make changes. But for this comment you must correct it.